src/components/SignUp.js

```diff
@@ -0,0 +1,143 @@
+import React, { Component } from 'react';
+import { Actions } from 'react-native-router-flux';
+import { ScrollView, Text, TextInput, TouchableOpacity } from 'react-native';
+
+class SignUp extends Component {
+  state = {
+     fname: '',
+     lname: '',
+     userName: '',
+     password: '',
+     email: '',
+     favGenres: [],
+     passwordLength: '',
+     passwordShort: true
+  }
+checkPassword(text) {
+  this.setState({ password: text });
+
+  if (text.length < 6) {
+    this.setState({
+      passwordLength: 'password must be at least 6 characters',
+      passwordShort: true
+    });
+  } else if (text.length < 25 && text.length > 5) {
+    this.setState({
+      passwordLength: this.state.passwordShort,
+      passwordShort: false
+  });
+
+  return this.state.passwordShort;
+  }
+}
+
+  render() {
+    return (
+      <ScrollView>
+        <Text style={styles.headerText}> Sign Up </Text>
+        <TextInput
+          placeholder="First Name"
+          style={styles.inputStyle}
+          onSubmitEditing={() => this.firstRef.focus()}
+          returnKeyType='next'
+          autoFocus
+          onChangeText={ (text) => this.setState({ fname: text })}
+          value={this.state.fname}
+        />
+
+        <TextInput
+          placeholder="Last Name"
+          style={styles.inputStyle}
+          ref={firstRef => this.firstRef = firstRef}
+          onSubmitEditing={() => this.lastRef.focus()}
+          returnKeyType='next'
+          onChangeText={(text) => this.setState({ lname: text })}
+          value={this.state.lname}
+        />
+
+        <TextInput
+          placeholder="Email"
+          style={styles.inputStyle}
+          ref={lastRef => this.lastRef = lastRef}
+          onSubmitEditing={() => this.emailRef.focus()}
+          returnKeyType='next'
+          onChangeText={ (text) => this.setState({ email: text })}
+          value={this.state.email}
+        />
+
+        <TextInput
+          placeholder="Username"
+          style={styles.inputStyle}
+          ref={emailRef => this.emailRef = emailRef}
+          onSubmitEditing={() => this.passwordRef.focus()}
+          returnKeyType='next'
+          onChangeText={(text) => this.setState({ userName: text })}
+          value={this.state.userName}
+        />
+
```

Comment:
The four `TextInput` fields above are extremely repetitive. This looks like something we can extract into a proper component, write the `TextInput` field logic once and populate the fields with props.

Thoughts?

Reply:
I see. We could create a FormField component that is similar to the Input component in Input.js:

```
const Input = ({ label, value, onChangeText, placeholder, secureTextEntry }) => {
  const { inputStyle, labelStyle, containerStyle } = styles;
  return (
    <View style={containerStyle}>
      <Text style={labelStyle}>{label}</Text>
      <TextInput
        secureTextEntry={secureTextEntry}
        autoCorrect={false}
        placeholder={placeholder}
        value={value}
        onChangeText={onChangeText}
        style={inputStyle}
      />
    </View>
  );
};

```

Reply:
I will fix this on signup_branch